src/test/java/core/basesyntax/service/RegistrationServiceImplTest.java

```diff
@@ -0,0 +1,74 @@
+package core.basesyntax.service;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+import static org.junit.jupiter.api.Assertions.fail;
+
+import core.basesyntax.exception.CustomException;
+import core.basesyntax.model.User;
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.Test;
+
+class RegistrationServiceImplTest {
+    private static RegistrationService service;
+
+    @BeforeAll
+    static void beforeAll() {
+        service = new RegistrationServiceImpl();
+    }
+
+    @Test
+    void userIsNull_notOk() {
+        User user = null;
+        try {
+            User register = service.register(user);
+        } catch (CustomException e) {
+            System.out.println("Custom exception was catched.");
+            return;
+        }
+        fail();
+    }
+
+    @Test
+    void userWithNullFields_notOk() {
+        User user = new User();
+        assertThrows(CustomException.class, () -> service.register(user));
+    }
+
+    @Test
+    void userWithNoValidData_notOk() {
+        User user = new User();
+        user.setId(1L);
+        user.setAge(14);
+        user.setLogin("login");
+        user.setPassword("password");
+        assertThrows(CustomException.class, () -> service.register(user));
```

Comment:
This test does not cover all the edge cases for invalid data. According to the checklist, you should add tests for all possible edge cases, such as a password with length exactly at the minimum required length.